crates/pallet-transaction-fees/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Minimum desired number of replicas of the blockchain to be stored by the network,
+        /// impacts storage fees.
+        #[pallet::constant]
+        type MinReplicationFactor: Get<u16>;
+
+        /// How much (ratio) of storage fees escrow should be given to farmer each block as a
+        /// reward.
+        #[pallet::constant]
+        type StorageFeesEscrowBlockReward: Get<(u64, u64)>;
+
+        /// How much (ratio) of storage fees collected in a block should be put into storage fees
+        /// escrow (with remaining issued to farmer immediately).
+        #[pallet::constant]
+        type StorageFeesEscrowBlockTax: Get<(u64, u64)>;
+
+        /// How many credits there is in circulation.
+        #[pallet::constant]
+        type CreditSupply: Get<BalanceOf<Self>>;
+
+        /// How much space there is on the network.
+        #[pallet::constant]
+        type TotalSpacePledged: Get<u64>;
+
+        /// How big is the history of the blockchain in archived state (thus includes erasure
+        /// coding, but not replication).
+        #[pallet::constant]
+        type BlockchainHistorySize: Get<u64>;
+
+        type Currency: Currency<Self::AccountId>;
+
+        type WeightInfo: WeightInfo;
+    }
+
+    /// Escrow of storage fees, a portion of it is released to the block author on every block
+    /// and portion of storage fees goes back into this pot.
+    #[pallet::storage]
+    #[pallet::getter(fn storage_fees_escrow)]
+    pub(super) type CollectedStorageFeesEscrow<T> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    /// Temporary value (cleared at block finalization) which contains cached value of
+    /// `TransactionByteFee` for current block.
+    #[pallet::storage]
+    pub(super) type TransactionByteFee<T> = StorageValue<_, BalanceOf<T>>;
+
+    /// Temporary value (cleared at block finalization) which contains current block author, so we
+    /// can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type BlockAuthor<T: Config> = StorageValue<_, T::AccountId>;
+
+    /// Temporary value (cleared at block finalization) which contains current block storage fees,
+    /// so we can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedStorageFees<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    /// Temporary value (cleared at block finalization) which contains current block compute fees,
+    /// so we can issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedComputeFees<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    /// Temporary value (cleared at block finalization) which contains current block tips, so we can
+    /// issue rewards during block finalization.
+    #[pallet::storage]
+    pub(super) type CollectedTips<T: Config> = StorageValue<_, BalanceOf<T>>;
```

Comment:
Since these 3 storage items are always processed together, some code can be simplified if we use just one storage with a struct:

```rust
pub struct CollectedFees<Balance> {
    storage: Balance,
    compute: Balance,
    tips: Balance,
}
```